TradeItIosTicketSDK2/TradeItLauncher.swift

```diff
@@ -19,19 +19,31 @@ protocol OAuthCompletionListener {
     override internal init() {}
 
     public func handleOAuthCallback(
-        onViewController viewController: UIViewController,
+        onViewController safariViewController: UIViewController,
         oAuthCallbackUrl: URL,
         onSuccessfulLink: ((TradeItLinkedBroker) -> Void)? = nil
     ) {
         print("=====> handleOAuthCallback: \(oAuthCallbackUrl.absoluteString)")
 
         let oAuthCallbackUrlParser = TradeItOAuthCallbackUrlParser(oAuthCallbackUrl: oAuthCallbackUrl)
 
-        self.oAuthCompletionUIFlow.presentOAuthCompletionFlow(
-            fromViewController: viewController,
-            oAuthCallbackUrlParser: oAuthCallbackUrlParser,
-            onSuccessfulLink: onSuccessfulLink
-        )
+        var parentViewController = safariViewController.presentingViewController
+        if parentViewController?.childViewControllers.first is TradeItSelectBrokerViewController {
+            parentViewController = parentViewController?.presentingViewController
+        }
+
+        guard let originalViewController = parentViewController else {
+            preconditionFailure("View hierarchy in unknown state.")
+        }
```

Comment:
This seems a little magic. I'm not entirely sure if it can end up in some other state.